Libraries/Core/ReactNativeVersionCheck.js

```diff
@@ -46,6 +47,8 @@ exports.checkVersions = function checkVersions(): void {
 function _formatVersion(version): string {
   return (
     `${version.major}.${version.minor}.${version.patch}` +
-    (version.prerelease !== null ? `-${version.prerelease}` : '')
+    (version.prerelease !== null && version.prerelease !== undefined
```

Comment:
In order to keep it less verbose, what do you think of checking for `version.prerelease != undefined` instead? By doing that, we can check if it's different from `null` and `undefined`

Reply:
Sure, I guess `version.prerelease &&` will serve the same purpose.. I wasn't sure if this would have be frowned upon

Reply:
Unfortunately no.
`version.prerelease` will be very different from simply `version.prerelease != undefined` because:

```javascript
const cond1 = version.prerelease;
const cond2 = !!version.prerelease;
const cond3 = (version.prerelease !== 0 && version.prerelease !== undefined && version.prerelease !== null && version.prerelease !== '' && version.prerelease !== false);
// cond1 === cond2 === cond3

const cond4 = version.prerelease != undefined;
const cond5 = version.prerelease != null;
const cond6 = version.prerelease !== undefined && version.prerelease !== null;
// cond4 === cond5 === cond6
```

Reply:
This is a minor issue. In the end, it will be a simplification of `version.prerelease !== null && version.prerelease !== undefined`

Reply:
I've updated, many thanks for taking the time to explain. I'm finding I've got 152 tests failing locally, and I'm not sure why, also flow isn't seeming to like what I've done. Is there anything obvious wrong with what I've done?